cql3/Cql.g

```diff
@@ -124,6 +124,21 @@ struct uninitialized {
     void check() const { if (!_val) { throw std::runtime_error("not intitialized"); } }
 };
 
+template <typename T>
+struct unwrap_uninitialized {
+    using type = T;
+};
+
+template <typename T>
+struct unwrap_uninitialized<uninitialized<T>> {
+    using type = T;
+};
+
+template <typename T>
+using unwrap_uninitialized_t = typename unwrap_uninitialized<T>::type;
+
+using uexpression = uninitialized<expression>;
+
 }
```

Comment:
I feel that this could use a comment, at first I didn't get how this is supposed to work.
Something like this would be helpful:
> At each step, the result of parsing is cast to `unwrap_initialized<T>::type`, which forces `unitialized<T>` values to be cast to `T`, which fails if the value is not initialized.